File: packages/shared/src/types/tenant.ts
export interface BusinessHours {
  [day: string]: {
    open: string;
    close: string;
    closed?: boolean;
  };
}

export interface TenantSettings {
  greeting_message?: string;
  notification_email?: string;
  notification_phone?: string;
  auto_reply_enabled?: boolean;
  follow_up_delay_hours?: number;
}

export interface Tenant {
  id: string;
  clerk_org_id: string;
  name: string;
  phone_number: string | null;
  retell_agent_id: string | null;
  business_hours: BusinessHours | null;
  settings: TenantSettings | null;
  created_at: string;
  updated_at: string;
}

File: apps/api/app/models/tenant.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, String, func
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Tenant(Base):
    __tablename__ = "tenants"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    clerk_org_id: Mapped[str] = mapped_column(String, unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)
    phone_number: Mapped[str | None] = mapped_column(String, nullable=True)
    retell_agent_id: Mapped[str | None] = mapped_column(String, nullable=True)
    business_hours: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    settings: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )
